test/integration/deploy_trigger_test.go

```diff
@@ -230,6 +222,27 @@ waitForPendingStatus:
 		t.Fatalf("expected deployment status %q, got %q", e, a)
 	}
 
+	// before we update the config, we need to update the state of the existing deployment
+	// this is required to be done manually since the deployment and deployer pod controllers are not run in this test
+	deployment.Annotations[deployapi.DeploymentStatusAnnotation] = string(deployapi.DeploymentStatusComplete)
+	// update the deployment
+	if _, err = kubeClusterAdminClient.ReplicationControllers(testutil.Namespace()).Update(deployment); err != nil {
+		t.Fatalf("Error updating existing deployment: %v", err)
+	}
+
+	event := <-watch.ResultChan()
+	if e, a := watchapi.Modified, event.Type; e != a {
+		t.Fatalf("expected watch event type %s, got %s", e, a)
+	}
```

Comment:
This integration test seems a bit fragile with the way its structured currently to use a watch. I will (either now or later as a separate PR) modify it to not use a watch for additions/modifications to replication controllers.